src/Pages/Detail/Aside.js

```diff
@@ -0,0 +1,289 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import styled, { css } from "styled-components";
+import { useHistory } from "react-router-dom";
+import {
+  AiOutlineHeart,
+  AiOutlineYoutube,
+  AiFillGift,
+  AiOutlineUser,
+  AiOutlineLike,
+  AiFillHeart,
+} from "react-icons/ai";
+import Button from "./Components/Button";
+import { Hr, API, Tooltip } from "./Components/Utils";
+
+const Aside = () => {
+  const [category, setCategory] = useState("");
+  const [creatorName, setCreatorName] = useState("");
+  const [classTitle, setClassTitle] = useState("");
+  const [price, setPrice] = useState("");
+  const [heart, setHeart] = useState(0);
+  const [heartToggle, setHeartToggle] = useState(false);
+  const [level, setLevel] = useState("");
+  const history = useHistory();
+
+  useEffect(() => {
+    axios.get(API).then((res) => setLevel(res.data.detail.class_info.level));
+  }, []);
+
+  const contentItem = [
+    [<AiOutlineYoutube size="18" />, <span>콘텐츠 이용권</span>],
+    [<AiFillGift size="18" />, <span>준비물 키트</span>],
+    [<AiOutlineUser size="18" />, <span>{level} 대상</span>],
+    [<AiOutlineLike size="18" />, <span>개인 만족도</span>],
+  ];
+
+  useEffect(() => {
+    axios.get(API).then((res) => {
+      setCategory(res.data.detail_aside.category);
+      setCreatorName(res.data.detail_aside.creator_name);
+      setClassTitle(res.data.detail_aside.title);
+      // setStatus(res.data.detail_aside.status);
+      setPrice(Number(res.data.detail_aside.price).toLocaleString("en"));
+      setHeart(res.data.detail_aside.heart);
+    });
+  }, []);
+
+  const classRequest = () => {
+    history.push("/detail/id/payment");
+  };
+
+  const heartButtonClick = () => {
+    if (!heartToggle) {
+      console.log(heart);
+      setHeart((prevState) => prevState + 1);
+      setHeartToggle(true);
+    } else {
+      const result = window.confirm(
+        "정말로 취소하시겠습니까? 알림 및 혜택을 받지 못하실 수 있습니다.",
+      );
+      if (result) {
+        setHeart((prevState) => prevState - 1);
+        setHeartToggle(false);
+      }
+    }
+  };
+
+  return (
+    <AsideWrap>
+      <div className="itemIntro">
+        <p>
+          {category} · {creatorName}
+        </p>
+        <h3>{classTitle}</h3>
+        <span className="nowRegi">
+          <div>바로 수강 가능</div>
+        </span>
+        <span className="installment">
+          <div>5개월 할부</div>
+          <h4>월 {price}원</h4>
+        </span>
+        <Hr />
+        <div className="contentinfo">
+          {contentItem.map((content) => (
+            <div>
+              {content[0]}
+              <span>{content[1]}</span>
```

Comment:
```suggestion
          {contentItem.map(([icon, desc]) => (
            <div>
              {icon}
              <span>{desc}</span>
```
- 배열도 구조분해할당 가능합니다. 구조분해 할당 하면서 인덱스 대신 변수명을 사용할 수 있으므로 적절한 이름을 지어줄 수 있습니다.
- 함수 파라미터 자리에서 직접 구조분해 할당 가능합니다.